node/src/tasks/bsp_upload_request.rs

```diff
@@ -0,0 +1,158 @@
+use std::path::Path;
+
+use forest_manager::traits::ForestStorage;
+use sc_tracing::tracing::{error, info, warn};
+use shc_common::types::HasherOutT;
+
+use file_manager::traits::{FileStorage, FileStorageWriteError, FileStorageWriteOutcome};
+use storage_hub_infra::event_bus::EventHandler;
+use tokio::{fs::File, io::AsyncWriteExt};
+
+use crate::services::{
+    file_transfer::{commands::FileTransferServiceInterface, events::RemoteUploadRequest},
+    handler::{StorageHubHandler, StorageHubHandlerConfig},
+};
+
+const LOG_TARGET: &str = "bsp-upload-request-handler";
+
+pub struct BspUploadRequestHandler<SHC: StorageHubHandlerConfig> {
+    storage_hub_handler: StorageHubHandler<SHC>,
+}
+
+impl<SHC: StorageHubHandlerConfig> Clone for BspUploadRequestHandler<SHC> {
```

Comment:
Why not making one single task file called `bsp_receive_file` whith two handlers, one for `RemoteUploadRequest` and one for `NewStorageRequest`. I think it would make it easier to follow the whole flow of that overarching task.